lang/java/reef-tests/src/main/java/org/apache/reef/tests/rack/awareness/RackAwareTask.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.tests.rack.awareness;
+
+import org.apache.commons.lang.Validate;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.task.Task;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+/**
+ * A simple task that receives the rack it is executed on
+ */
+public final class RackAwareTask implements Task {
```

Comment:
I don't think we need to test this all the way to a `Task`. If the handler for `AllocatedEvaluator` gets the right rack, we should be good. That simplifies the test a lot.